### DISCUSSION TITLE
balancing income paths

### DISCUSSION THREAD

**Discussion by flowers9** [endless-sky/endless-sky#9041] [Balance, Mechanics, and Interactions]
Currently, there are roughly four ways to produce income in the game: completing jobs/missions, trading commodities, mining, and combat (looting and capturing ships).  There's some overlap, as some missions involve combat.

These four approaches offer wildly different income profiles, and strongly incentivize knowledgeable players to focus on one particular approach at a time (the one with the highest return).  Further, the unbalanced nature makes it difficult to pace and place content, particularly in human space which can be quickly left behind, despite the large amount of content available.

So, here's a look at early game balancing, with hooks for both early and (hopefully) later game balancing as well.

Early game, mining strongly dominates, with a starting ~$230k/day income and a doubling time of 2 days, up to a maximum of ~$1.6m/day.  Commodity trading on that level requires an investment of about $150m, and in general has a doubling time of ~82 days (but no upper limit).  Combat, once you can afford the ship(s) and get a high combat rating, can offer returns similar to mining, but the return is fairly flat once you get there, also like mining (I didn't look into the whole combat income procession to that level).  Non-combat jobs offer far less, perhaps $60k/day if you're very lucky.

I would like to see the non-job income brought more in line with the job-based income, in order to give players more options they feel worth taking and to encourage taking jobs, with their higher likelihood of engaging with content, while leaving the other options as viable (but not dominating) approaches and as decent filler while doing jobs.  To this end, I would suggest modifying all approaches.

### Mining:

The primary limit on mining is the total value of mineables in a system.

* Reduce the value of mineables to 1/10 their current value.
* Introduce a mining rating that increases with mining (destroying asteroids), and increases the amount of mineables produced when destroying an asteroid.
* Have (more) mining weapons, and give them a mining efficiency rating that can increase the amount of mineables produced when destroying an asteroid.  (For fun, some could trade off mining speed and efficiency.)

### Commodity Trading:

Currently, the only limit on trading is the size of your cargo hold.

* Put a limit on the amount of a given commodity available at a given time to 2000 (possibly plus 1/100 the system supply, to give it some randomness (supply seems to range about -30k to 30k)).
* Introduce a trading rating that increases when buying commodities, that increases the limit.

### Capturing Ships:

The primary limit on capturing ships is the crew loss attacking and crewing the captured ship, and the value of the ships.

* Put defenders (automated defense) on uncrewed ships, and give them a meaningful prize-crew requirement (lore-wise, the prize-crew is having to run the ship from outside the ship after they destroy the AI running it that was trying to kill them).
* Have a crew mortality rating that tracks crew death history and influences crew salaries (akin to how credit rating tracks missing payments and influences mortgage rates).
* Have (active) fleet worth influence crew salaries.

The objective is that crew salaries can rise to the point of putting a substantial dent in the revenue from capturing ships.  Tracking by fleet worth means it can be relevant even in the later game.

### Jobs:

* Introduce jobs that require mining or buying commodities as part of the mission, just as some jobs currently require combat.  Essentially, they'd be "deliver X to Y", but unlike current jobs they wouldn't provide X, leaving the player to find places to mine it, or cheap places to buy it.

This helps the new ratings track player progress, much as combat rating does now.  All the ratings, and their effects, can be tweaked to different stages of game progress (or so I hope ;).  In particular, they can be used to slow early game income from non-job approaches to prevent jobs from being quickly left behind, and they can offer the player a sense of progression from play that might otherwise be a bit boring.

(#7827 mentions similar issues to this, but I think this offers a different take.  Plus, that conversation ended six months ago and I didn't want to necro the thread.  Thanks to @movingpictures for the mortality rating idea.)

<details>
<summary>Income details (or how I came up with those numbers) (spoilers)</summary>

<details>
<summary>Mining</summary>
Caph offers about $1.6m in mineables, sells Sparrows and cargo expansions (and asteroid scanners), and is 4 hops from the default start location.  Strip the shield generator, battery pack, and hyperdrive and put in two cargo expansions and a supercapacitor.  Add an asteroid scanner after day 1.  Take out loans when available, and buy new Sparrows with the same configuration (sans scanner).  Once you hit the system mineables limit (~450 tons cargo), you can swap out cargo expansions for beam lasers to speed up mining and eliminate any pirate threat.  Repay the mortgages once you've maxed out (which only takes a couple of weeks).
</details>

<details>
<summary>Trading</summary>
At Mirfak, you can buy Star Barges, strip out everything but the hyperdrive, add an X1050 engine, an AA fuel cell, and 3 fuel pods.  To avoid pirates, park them, jump to Hai Due, unpark them and wait for them to show up.  Then swap out the fuel pods and AA fuel cell for a chasm battery and cargo expansions, giving 110 tons cargo for $387k.  In Hai space (no pirates), you can do one-hop commodity trading with a profit margin of ~$100/ton, yielding $50/ton/day.
</details>

<details>
<summary>Combat</summary>
In human space, the most ridiculous I've seen is jumping between Betelgeuse and Alheka with a 6+ combat rating (and a ~$40m combat fleet).  3-5 combat missions per planet offer $300k-$500k each (and often objectives for several missions will show up at the same time), plus you can capture many of the ships, and both systems have shipyards you can sell the captured ships at (assuming you don't want to just build up your fleet).
</details>

<details>
<summary>Jobs</summary>
The best non-combat mission I've seen is a one-jump to deliver space-divers for $120k.  In general, it can be very hard to line up high paying jobs such that there's not a lot of jumping involved.
</details>
</details>

**Comment by Zitchas** [comment]
This is definitely something that comes up from time to time, and there's no clear agreed upon solutions. That being said, it's good to see it documented and contrasted.

My general pattern - if I'm going to maximize wealth - is mining for the early game, then combat once I'm in a good fighting ship, then commodity trading once I can get a fleet to really push it.

My typical pattern, as I generally don't care about maximizing wealth, is jobs, then combat supported by jobs, then missions combined with combat, then missions with occasional jobs or combat. From mid game onwards I rarely do anything that is purely wealth focused.

**Comment by Hurleveur** [comment]
Here are my thoughts about the 4 ways and how we could balance them:
- Trade: I think it'd make sense to make mass trade harder rather than punishing small trders. We could make trade fluctuate faster than it currently does, for instance.
- Jobs: I think you forgot the uhai ones that unlock after you sell them JDs - we shouldnt just nerf the income of jobs tho I think, but rather make them more interesting (harder to complete for instance)
- Mining: we could do with a few very good systems like the supernova ones (or mb those could have better yields) but I think that the Gegno will improve late game mining
- Fighting: I think looting is fine, and as for capping there is a pr for making it not so profitable, but I think the best way to address this would be to have the ppl you made mad come after you (mb as bounty hunters, or korath/uhai interdiction fleets) before you can snowball too much, forcing you to run away for a while. Also specifically for bounty hunters its too easy to cap them - it should not reward you money when you do so, and we could do with a new self destruct that happens when the enemy crew gets desperate.
- Big fleets: we should have something to decentivize those, but not a hard cap (like say just have higher crew costs the more ships you have). at the very least ship captains should cost more, somehow depending on how much crew they must manage.

**Comment by mOctave** [comment]
When it comes to trade, maybe all planets should actually be given official populations so we can better model their economies? Just a thought.
